### PR TITLE
AreaSeries animation rework.

### PR DESCRIPTION
**Pull request by alantreadway**
Reworks the AreaSeries + RangeAreaSeries to use the new `fromToMotion`-based animation mechanisms, and tries to reuse some of the `LineSeries` logic to achieve the desired animations.

Bonus:
- Splits re-usable parts of `pathUtil.ts` into other util modules.
- Improves area-series node-data generation readabliity.
- Fixes `LineSeries` `dimOpacity` handling that was broken in it's rework PR.

### CODE REVIEW

File: packages/ag-charts-community/src/chart/series/cartesian/__image_snapshots__/area-series-test-ts-area-series-create-for-area-category-x-axis-fractional-log-y-axis-it-should-render-to-canvas-as-expected-1-snap.png
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/cartesian/__image_snapshots__/area-series-test-ts-area-series-create-for-area-category-x-axis-fractional-log-y-axis-it-should-render-to-canvas-as-expected-1-snap.png:1]: I'm working on fixing these 6 snapshots for `AreaSeries` animations - however there is a small chance they are rendering incorrectly due to the fact we don't currently fully-mock `requestAnimationFrame`, which may require follow-up work to address.

File: packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts [hunk 1]
```diff
@@ -30,400 +32,133 @@ export type AreaPathDatum = {
...
-                delay: seriesRect?.width ? (datum.x / seriesRect.width) * duration : 0,
-                duration: markerDuration,
-                onUpdate(opacity) {
-                    label.opacity = opacity;
-                },
-            });
-        });
-    });
+        previousTo = to;
+    }
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts]: This looks the same as the `render()` function inside `prepareLinePathAnimation()` in `lineUtils`. Is it possible to reuse this at all?

Comment (lsjroberts) [reply packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts]: Well, that got turned into `renderPartialLine()` in this diff, but still looks the same as this

File: packages/ag-charts-enterprise/src/series/range-area/__image_snapshots__/range-area-test-ts-chart-initial-animation-for-range-area-options-should-animate-at-25-1-snap.png
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/series/range-area/__image_snapshots__/range-area-test-ts-chart-initial-animation-for-range-area-options-should-animate-at-25-1-snap.png:1]: These look like regressions?

**Review by lsjroberts — COMMENTED** [review]
Looks great, couple of comments.

File: packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts]: Very good shout - they are identical, I think they were different at some intermediate stage but they turned out to just be a general 'animated-path' rendering - I've renamed and moved to `pathUtil.ts`.

File: packages/ag-charts-enterprise/src/series/range-area/__image_snapshots__/range-area-test-ts-chart-initial-animation-for-range-area-options-should-animate-at-25-1-snap.png
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/series/range-area/__image_snapshots__/range-area-test-ts-chart-initial-animation-for-range-area-options-should-animate-at-25-1-snap.png:1]: The series-labels? So I've used the existing series-label animation primitives we use everywhere for consistency, I think that having them fade-in at the end is fine, and we should change the behaviour across all series if there is a better approach?

Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/series/range-area/__image_snapshots__/range-area-test-ts-chart-initial-animation-for-range-area-options-should-animate-at-25-1-snap.png:1]: Ahh, of course, yeah no worries

**Review by lsjroberts — APPROVED** [review]

**Review by alexanderby — APPROVED** [review]

File: packages/ag-charts-community/src/chart/series/cartesian/__image_snapshots__/area-series-test-ts-area-series-create-for-area-category-x-axis-fractional-log-y-axis-it-should-render-to-canvas-as-expected-1-snap.png
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/series/cartesian/__image_snapshots__/area-series-test-ts-area-series-create-for-area-category-x-axis-fractional-log-y-axis-it-should-render-to-canvas-as-expected-1-snap.png:1]: Fixed - this was a legitimate break with `LogAxis` cases!

**Comment by alantreadway** [comment]
@lsjroberts PTAL

**Review by lsjroberts — APPROVED** [review]

**Review by alexanderby — APPROVED** [review]

**Pull request merged**